test/e2e/tests/load_balancing.go

```diff
@@ -0,0 +1,295 @@
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			headers := []string{"0.0.0.0", "1.2.3.4", "4.5.6.7", "7.8.9.10", "10.11.12.13"}
+
+			for _, header := range headers {
+				// Same test header will always hit the same endpoint.
+				var expectPodName string
+
+				for i := 0; i < sendRequests; i++ {
+					req.Headers["Lb-Test-Header"] = []string{header}
+					cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+					if err != nil {
+						t.Errorf("failed to get expected response: %v", err)
+					}
+
+					if err := http.CompareRequest(t, &req, cReq, cResp, expectedResponse); err != nil {
+						t.Errorf("failed to compare request and response: %v", err)
+					}
+
+					podName := cReq.Pod
+					if len(podName) == 0 {
+						// it shouldn't be missing here
+						t.Errorf("failed to get pod header in response: %v", err)
+					} else {
+						if len(expectPodName) == 0 {
+							expectPodName = podName
+						} else {
+							require.Equal(t, expectPodName, podName)
+						}
+					}
+				}
+			}
+		})
+	},
+}
+
+var ConsistentHashCookieLoadBalancingTest = suite.ConformanceTest{
+	ShortName:   "Cookie based Consistent Hash Load Balancing",
+	Description: "Test for cookie based consistent hash load balancing type",
+	Manifests:   []string{"testdata/load_balancing_consistent_hash_cookie.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		const sendRequests = 10
+
+		ns := "gateway-conformance-infra"
+		routeNN := types.NamespacedName{Name: "cookie-lb-route", Namespace: ns}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+
+		ancestorRef := gwapiv1a2.ParentReference{
+			Group:     gatewayapi.GroupPtr(gwapiv1.GroupName),
+			Kind:      gatewayapi.KindPtr(gatewayapi.KindGateway),
+			Namespace: gatewayapi.NamespacePtr(gwNN.Namespace),
+			Name:      gwapiv1.ObjectName(gwNN.Name),
+		}
+		BackendTrafficPolicyMustBeAccepted(t, suite.Client, types.NamespacedName{Name: "cookie-lb-policy", Namespace: ns}, suite.ControllerName, ancestorRef)
+
+		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+
+		t.Run("all traffics route to the same backend with same cookie", func(t *testing.T) {
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/cookie",
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			cookies := []string{"0.0.0.0", "1.2.3.4", "4.5.6.7", "7.8.9.10", "10.11.12.13"}
+
+			for _, cookie := range cookies {
+				// Same test cookie will always hit the same endpoint.
+				var expectPodName string
+
+				for i := 0; i < sendRequests; i++ {
+					req.Headers["Set-Cookie"] = []string{fmt.Sprintf("Lb-Test=%s", cookie)}
```

Comment:
It seems that gatewayapi conformance test suite does not support setting cookie for one request, and this cookie-setting way is ref to `http.WriteCookie` in `net/http` pkg, but unfortunately, not working as well.